src/lib/snark_worker/functor.ml

```diff
@@ -95,8 +97,26 @@ module Make (Inputs : Intf.Inputs_intf) :
     end
   end
 
+  module DumpedWorkSpec = struct
+    type t =
+      { prover : Signature_lib.Public_key.Compressed.Stable.V1.t
+      ; spec : Work.Single.Spec.t One_or_two.Stable.V1.t
+      ; fee : Currency.Fee.Stable.V1.t
+      }
+    [@@deriving yojson]
+  end
+
   let perform (s : Worker_state.t) public_key
       ({ instances; fee } as spec : Work.Spec.t) =
+    let dir = Sys.getenv_exn "PATH_DUMP_SNARK_WORK_SPEC" in
+    let to_dump : DumpedWorkSpec.t =
+      { prover = public_key; spec = spec.instances; fee = spec.fee }
+    in
+    work_spec_counter := !work_spec_counter + 1 ;
```

Comment:
As long as snark worker is single threaded, we don't have race condition writing to same file here.